Venmo/src/main/java/com/braintreepayments/api/VenmoActivityResultContract.java

```diff
@@ -0,0 +1,83 @@
+package com.braintreepayments.api;
+
+import android.content.ComponentName;
+import android.content.Context;
+import android.content.Intent;
+
+import androidx.activity.result.contract.ActivityResultContract;
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.annotation.VisibleForTesting;
+import androidx.appcompat.app.AppCompatActivity;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+
+class VenmoActivityResultContract extends ActivityResultContract<VenmoIntentData, VenmoResult> {
+
+    static final String VENMO_PACKAGE_NAME = "com.venmo";
+    static final String APP_SWITCH_ACTIVITY = "controller.SetupMerchantActivity";
+    static final String META_KEY = "_meta";
+    static final String EXTRA_MERCHANT_ID = "com.braintreepayments.api.MERCHANT_ID";
+    static final String EXTRA_ACCESS_TOKEN = "com.braintreepayments.api.ACCESS_TOKEN";
+    static final String EXTRA_ENVIRONMENT = "com.braintreepayments.api.ENVIRONMENT";
+    static final String EXTRA_BRAINTREE_DATA = "com.braintreepayments.api.EXTRA_BRAINTREE_DATA";
+    static final String EXTRA_PAYMENT_METHOD_NONCE = "com.braintreepayments.api.EXTRA_PAYMENT_METHOD_NONCE";
+    static final String EXTRA_USERNAME = "com.braintreepayments.api.EXTRA_USER_NAME";
+    static final String EXTRA_RESOURCE_ID = "com.braintreepayments.api.EXTRA_RESOURCE_ID";
+
+    @VisibleForTesting
+    boolean shouldVault;
```

Comment:
We still might need to store this in shared prefs. `createIntent` may not be called after a process kill and this value could be `false` when the merchant intends it to be `true`.

Reply:
Ah yeah good call - we moved this back to sharedPrefs.